core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -83,6 +89,11 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
   def main(args: Array[String]) {
 
     info("Starting mirror maker")
+    println("WARNING: The default partition assignment strategy of mirror maker will change from 'range' to 'roundrobin' " +
```

Comment:
@vahidhashemian To clarify a little. When you start the MM process (by executing the MM command), one consumer instance will be created. Other MM processes (potentially running on different hosts) will have their own consumer instance. When multiple MM processes share the same consumer group Id, they will work in concert as a single consumer group in order to balance the load. The problem is that they must all agree on the assignment protocol. In a rolling upgrade scenario, you will have some MM processes running on some hosts which still use the old config, and you will have some MM processes running on other hosts which use the newer config. So what's the behavior when some MM hosts are supporting 'range' and some are supporting 'round-robin'? In general, it just doesn't work, but there is an option available for the new consumer (as I described above). Does that make sense?